src/rust/engine/fs/src/directory.rs

```diff
@@ -311,6 +314,43 @@ impl From<&File> for remexec::FileNode {
   }
 }
 
+#[derive(Clone, Debug, DeepSizeOf)]
+pub struct Symlink {
+  name: Name,
+  target: String,
```

Comment:
Maybe `target` should be typed as `RelativePath` or `PathBuf`? `RelativePath` is a Pants concept that enforces that the path is a relative path.

For now, I don't believe we need to support absolute paths (other than maybe `/dev/null`).

Reply:
Went with `RelativePath`